### PR TITLE
Clarify apply_ufunc error message (Issue #2078)

### PR DESCRIPTION
**Pull request by rdrussotto**
- [x] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API

Changed an error message thrown by a dependent function of apply_ufunc to specify that "an input variable" rather than "the input variable" did not have the correct dimensions. This error occurs when apply_ufunc is used on a Dataset with multiple variables, at least one of which does not have correct dimensions matching the other input arguments. Wanted to add variable name to error message to specify which variable is causing the error, but this isn't possible without significantly refactoring the code.

### CODE REVIEW

File: xarray/core/computation.py [hunk 1]
```diff
@@ -482,9 +482,9 @@ def broadcast_compat_data(variable, broadcast_dims, core_dims):
     missing_core_dims = [d for d in core_dims if d not in set_old_dims]
     if missing_core_dims:
         raise ValueError(
-            'operand to apply_ufunc has required core dimensions %r, but '
-            'some of these are missing on the input variable:  %r'
-            % (list(core_dims), missing_core_dims))
+            'operand to apply_ufunc has required core dimensions {}, but '
+            'some of these are missing on an input variable: {}'.format(
```
Comment (dcherian) [inline xarray/core/computation.py]:
```suggestion
            'some of these dimensions are absent on an input variable: {}'.format(
```

**Comment by dcherian** [comment]
Thanks @rdrussotto. Do you like the suggested change? If so, commit that and I'll merge.

**Comment by dcherian** [comment]
The test failure should be fixed by merging master.

**Comment by rdrussotto** [comment]
I agree with the suggested change, just committed it.

**Comment by max-sixty** [comment]
Thanks @rdrussotto - there are now a couple of small merge conflicts - could you resolve? Then we can merge

**Comment by rdrussotto** [comment]
I tried to resolve the merge conflicts but given the tests don't pass I might have messed something up...

**Comment by dcherian** [comment]
You just need to run `black xarray/core/computation.py`

**Comment by dcherian** [comment]
Yay! Thanks @rdrussotto

**Pull request merged**